### PR TITLE
Documentation for `asoctl clean crds`

### LINKED ISSUE #2858: Document asoctl crd clean process for v2.0.0
We need to update the asoctl documentation with a good description of what it does and how to use it, and _why_ it should be used.

Can highlight the error that you get if you _dont_ run this tool.
```
E0405 23:48:54.061666       1 main.go:124] setup "msg"="failed to apply CRDs" "error"="failed to apply CRD storageaccountsqueueservicesqueues.storage.azure.com: CustomResourceDefinition.apiextensions.k8s.io \"storageaccountsqueueservicesqueues.storage.azure.com\" is invalid: status.storedVersions[0]: Invalid value: \"v1alpha1api20210401storage\": must appear in spec.versions" 
```

### PR DESCRIPTION
**Pull request by super-harsh**
Closes #2858 

**What this PR does / why we need it**:

Adds documentation for `asoctl clean crds` and usage.

**If applicable**:
- [x] this PR contains documentation
- [ ] this PR contains tests

### CODE REVIEW

File: docs/hugo/content/tools/asoctl.md [hunk 1]
```diff
@@ -29,7 +29,37 @@ Available Commands:
...
+
+`--dry-run` flag outputs about CRDs and CRs to be updated and **does not** modify any CRD and CRs.
+
+### Steps for migration using `asoctl clean crds`:
+
+- Ensure the current ASOv2 version is `beta.5`
+- Run `asoctl clean crds`
+- Upgrade ASOv2 to `2.0.0`
+
+Using `asoctl clean crds` is an important step if `v1alpha1api` resources are present in the cluster. If not used correctly, operator pod would output log like:
```
Comment (theunrepentantgeek) [inline docs/hugo/content/tools/asoctl.md:58]:
```suggestion
Using `asoctl clean crds` is an important step if `v1alpha1api` resources have ever been present in the cluster. If not used correctly, operator pod will produce log error messages:
```

File: docs/hugo/content/tools/asoctl.md [hunk 2]
```diff
@@ -29,7 +29,37 @@ Available Commands:
...
+      --verbose   Enable verbose logging
+```
+
+`--dry-run` flag outputs about CRDs and CRs to be updated and **does not** modify any CRD and CRs.
+
+### Steps for migration using `asoctl clean crds`:
+
+- Ensure the current ASOv2 version is `beta.5`
+- Run `asoctl clean crds`
+- Upgrade ASOv2 to `2.0.0`
```
Comment (theunrepentantgeek) [inline docs/hugo/content/tools/asoctl.md:56]: Readers may not understand that you're showing a command line.

```suggestion
**Prerequisite:** Ensure the current ASO v2 version in your cluster is `beta.5`.

Run the migration tool:
``` bash
$ asoctl clean crds
```

Once that's successfully run, you can upgrade ASO to `v2.0.0`
```

Maybe include some sample output for the command?

**Review by theunrepentantgeek — APPROVED** [review]

File: docs/hugo/content/tools/asoctl.md (hunk 2 above)
Comment (super-harsh) [reply docs/hugo/content/tools/asoctl.md]: Done!

**Comment by super-harsh** [comment]
TODO: Update new logging style in docs

File: docs/hugo/content/tools/asoctl.md [hunk 3]
```diff
@@ -26,7 +26,59 @@ Available Commands:
 
 ## Clean CRDs
 
-TBC
+This command can be used to prepare ASOv2 `v1alpha1api`(deprecated in v2.0.0) CustomResources and CustomResourceDefinitions for ASO `v2.0.0` release. 
+It ensures that any ASOv2 `v1alpha1api` deprecated version resources that may have been stored in etcd get migrated to `v1beta` version before upgrading ASO to `v2.0.0`.
```
[the remark this answers is not in the document]
Comment (super-harsh) [reply docs/hugo/content/tools/asoctl.md:30]: Have mentioned something like that below with the error log. Do you think we should put something similar here at the top?

**Pull request merged**